File: server/ruby/app/infrastructure/mappers/payment_details/acquirer_information_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class AcquirerInformationMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::AcquirerInformation.new
          dto.acquirer_selection_information = Infrastructure::Mappers::PaymentDetails::AcquirerSelectionInformationMapper.from_sdk_response(response.acquirer_selection_information)
          dto.name = response.name

          dto
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/acquirer_information_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.acquirer_information import AcquirerInformation
from app.infrastructure.mappers.payment_details.acquirer_selection_information_mapper import AcquirerSelectionInformationMapper
from onlinepayments.sdk.domain.acquirer_information import AcquirerInformation as AcquirerInformationSdk

class AcquirerInformationMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[AcquirerInformationSdk]) -> Optional[AcquirerInformation]:
        if response is None:
            return None

        dto = AcquirerInformation()
        dto.acquirer_selection_information = AcquirerSelectionInformationMapper.map_from_sdk_response(response.acquirer_selection_information)
        dto.name = response.name

        return dto